dimod/higherorder/polynomial.py

```diff
@@ -134,11 +134,11 @@ def __eq__(self, other):
         return (
             self.vartype == other.vartype
             and all(
-                (not bias or other_terms[term] == bias)
+                (not bias or other_terms.get(term) == bias)
```

Comment:
While it's not really an intended use case, I suppose users may want to specify `None` as a bias. Probably better to have a special signal object for this.

Reply:
I am not sure I follow, `.get` returns None if no bias is specified. We could make that explicit with `.get(term, None)` or specify a different value, as long is it is false, so that `other_terms.get(term) == bias` is still guaranteed to be false. Either way the actual result of `.get` is invisible to the user though.

Reply:
`.get(term, 0.)` seems the most correct mathematically

Reply:
That makes sense to me

Reply:
Done